ui/app/components/auth/form/base.ts

```diff
@@ -53,28 +57,25 @@ export default class AuthBase extends Component<Args> {
   }
 
   login = task(
-    waitFor(async (data) => {
+    waitFor(async (formData) => {
       try {
         const authResponse = await this.auth.authenticate({
           clusterId: this.args.cluster.id,
           backend: this.args.authType,
-          data,
+          data: formData,
           selectedAuth: this.args.authType,
         });
 
-        // responsible for redirect after auth data is persisted
-        this.onSuccess(authResponse);
+        this.handleAuthResponse(authResponse);
       } catch (error) {
         this.onError(error as Error);
       }
     })
   );
 
-  // if we move auth service authSuccess method here (or to each auth method component)
-  // then call that before calling parent this.args.onSuccess
-  onSuccess(authResponse: object) {
-    //  responsible for redirect after auth data is persisted
-    this.args.onSuccess(authResponse, this.args.authType);
+  handleAuthResponse(authResponse: AuthData) {
+    // calls onAuthResponse in parent auth/page.js component
+    this.args.onSuccess(authResponse);
   }
```

Comment:
Could `this.args.onSuccess(authResponse)` be called directly from the login method instead?

Reply:
Good question! This method will eventually contain logic to manage MFA scenarios, something like:
```js
  // this.args.onSuccess calls onAuthResponse in auth/page.js component
  handleAuthResponse(authResponse: AuthData, authType?: string) {
    if (Object.keys(authResponse).includes('mfa_requirement')) {
      const parsedMfaAuthResponse = this.auth._parseMfaResponse(authResponse.mfa_requirement);
      this.args.onSuccess(parsedMfaAuthResponse, authType);
    } else {
      this.args.onSuccess(authResponse);
    }
  }